src/formatDate.js

```diff
@@ -8,7 +8,30 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[fromFormat.length - 1];
+  const newSeparator = toFormat[toFormat.length - 1];
+
+  const oldDateParts = date.split(oldSeparator);
+
+  const dateComponents = fromFormat.slice(0, -1).reduce((acc, key, i) => {
+    acc[key] = oldDateParts[i];
+
+    return acc;
+  }, {});
+
+  if (dateComponents['YY']) {
+    dateComponents['YYYY'] =
+      dateComponents['YY'] < 30
+        ? '20' + dateComponents['YY']
+        : '19' + dateComponents['YY'];
```

Comment:
The logic for converting from 'YY' to 'YYYY' is correct, but it assumes that 'YY' will always be present in the `dateComponents` object. If 'YY' is not a part of the `fromFormat`, this code will create an unnecessary 'YYYY' key in `dateComponents`. It would be more robust to check if the 'YYYY' key exists in the `fromFormat` and then perform the conversion based on that.